### PR TITLE
feat: .alva files

### LINKED ISSUE #19: Create new project
As a user I want to create a project inside a styleguide.

### LINKED ISSUE #199: Disable drop zone for "molecules" that can't have child elements
The examples provide a molecule called teaser and is possible to create children for that element, but that has no effect on the preview. 

Feature request:  
If an element can't have children remove the ability to drag and drop other elements onto it, so it is not possible to create misleading structures

### LINKED ISSUE #440: Handle Projects as .alva Files
tbd

### PR DESCRIPTION
**Pull request by marionebl**
## Rationale

This PR introduces a changed persistence system where everything regarding a Alva project is saved into a single, shareable `.alva` file. For the time being this file is a `.yml` file with the `.alva` file extension. The intended improvements for our users are

* Files that can be shared easily 
* No need to do technical setup on a component library for trying out alva
* Better support for first steps without a own component library

## Done
- [x] Graph: Decouple pattern-property from pattern
- [x] Graph: Decouple element-property from element
- [x] Graph: Decouple element from content
- [x] Graph: Decouple pattern-property from pattern-property
- [x] Restore component search
- [x] Reintroduce component library analyzer as "component import"
- [x] Fix selected element highlighting in preview
- [x] Remove notion of multiple projects
- [x] Decouple styleguide from analyzer
- [x] Add `toJSON` and `static from` methods to all classes for serialization/deserialization
- [x] Move all interaction with the file system to the main process
- [x] Add full typings for messages between render process, main and preview
- [x] Model slots in elements as distinct entity: `ElementContent`
- [x] Model properties in elements as distinct entity: `ElementProperty`

## Dropped

- [ ] ~~Remove unused model and store methods~~
- [ ] ~~Associate the .alva file extension with Alva~~

## Notes
- There is weak encapsulation between the Alva data structures and the preview renderer. Hardening this boundary would result in a sturdier system. The client-side preview API should be powered by a  data structure initializied via `Project.from`
- Moving to a graph where all data structures are loosely coupled via uuids would make most changesets smaller by avoiding sprawling, tree-spanning changes. We might consider using a graph modelling library to offload the infrastructure work for this
- `element-property` should be split into value-specific classes analogous to `pattern-property`
- Consider adding a purge functionality that collects orphaned elements from the project registry
- Consider splitting `ElementLocationCommand` into smaller abstractions
- Consider abstracting away the `*Command` pattern inside the remaining ViewStore

Fixes #440 
Fixes #19 
Fixes #199

### CODE REVIEW

File: src/preview/preview-renderer.tsx [hunk 1]
```diff
@@ -140,10 +149,12 @@ class PreviewComponent extends React.Component<PreviewComponentProps> {
 		const props = this.props as InjectedPreviewComponentProps;
 		if (props.uuid === props.store.elementId) {
 			const node = ReactDom.findDOMNode(this);
+			console.log('&&&&&&&&&&&&', props.store.elementId);
 			if (node) {
 				props.highlight.setSize(node as Element);
 			}
 		}
+		console.log('&&&&&&&&&', props.highlight.node);
```
Comment (tilmx) [inline src/preview/preview-renderer.tsx]: Should we remove the console logs? @marionebl

File: package.json [hunk 2]
```diff
@@ -106,13 +106,18 @@
...
     "@types/mime-types": "2.1.0",
     "@types/mock-fs": "3.6.30",
     "@types/node": "9.6.5",
     "@types/object-path": "0.9.29",
     "@types/react": "16.3.10",
     "@types/react-dom": "16.0.5",
+    "@types/react-helmet": "5.0.6",
+    "@types/read-pkg": "3.0.0",
+    "@types/readdir-enhanced": "2.2.0",
     "@types/smoothscroll-polyfill": "0.3.0",
```
Comment (tilmx) [inline package.json]: We should uninstall this as we do not need it any more @marionebl

**Comment by markusoelhafen** [comment]
Currently it's not possible to duplicate elements via Cmd+D / Ctrl+D

**Comment by marionebl** [comment]
> Currently it's not possible to duplicate elements via Cmd+D / Ctrl+D

Fixed via https://github.com/meetalva/alva/pull/454/commits/1f9ce1e66e4f5f03adfdf26b0d05c49be0bf42e4

**Pull request merged**